src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -848,6 +848,18 @@ leader_balancer::build_preference_index() {
         const auto& preference
           = topic.second.get_configuration().properties.leaders_preference;
         if (preference.has_value()) {
+            if (
+              should_sanction
+              && preference.value() != config::leaders_preference{}) {
```

Comment:
I wonder if we should use the default value?
```suggestion
              && preference.value()
                   != config::shard_local_cfg()
                        .default_leaders_preference.default_value()) {
```

Reply:
Once we are done with this work, this logic will be duplicated in here, in `feature_manager::report_enterprise_features` and in `get_enterprise_features` (both overloads).

We currently have the logic of what is a restricted value in the `enterprise` property. Given that this is a topic property that refers to the same thing, it feels like we may need to re-evaluate our design down the line, when we have seen more use cases. (Note that we have enterprise features that reflect node config, as well + some enterprise features don't match the reporting we want to be doing)

My gut feeling is that we should use the same notation in all of them, so it is less probable to forget about them if we get to refactor this. If we want to use the property instead of the pre-existing notation, i am happy to change all of the above to
```
if (config::shard_local_cfg().default_leaders_preference.check_restricted(value) {...}
```

Reply:
> If we want to use the property instead of the pre-existing notation, i am happy to change all of the above to

Ahh, `check_restricted` feels right to me, too.

Reply:
ok, i'll change it everywhere (and it will be easier to find all the instances if we come around to refactor) :+1: